apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -766,6 +766,30 @@ defmodule Astarte.AppEngine.API.Device do
     end
   end
 
+  defp cast_value(:longinteger, string_value) when is_binary(string_value) do
+    case Integer.parse(string_value) do
+      {int_value, ""} ->
+        {:ok, int_value}
+
+      _ ->
+        {:error, :unexpected_value_type, expected: :longinteger}
+    end
+  end
+
+  defp cast_value(:longinteger, _string_value) do
```

Comment:
You're missing a clause above this one:
```elixir
defp cast_value(:longinteger, int_value) when is_integer(int_value) do
  {:ok, int_value}
end
```
Otherwise we would stop accepting integer values, which are valid


Also, small nit: in the latest clause (the one I'm commenting on) the value is actually _not_ a string, so just call the unused variable `_value`.